### DISCUSSION TITLE
Loading gltf with a custom MaterialProvider

### DISCUSSION THREAD

**Discussion by jnmaloney** [google/filament#4250] [Q&A]
I have a bunch of models and animations that I want to import, at the moment I'm using gltf format. I also want to use a material library of premade materials and apply them to the imported model by material name. 

I've tried to use a custom gltfio::MaterialProvider that returns the named material instance. The problem is that it tries to apply parameters that don't exist like emissiveFactor that cause a material panic.

I'm not sure if I'm doing something wrong. Is it possible to do this with loading gltf models?

**Comment by romainguy** [comment]
I'll let @prideout answer but alternatively you can change the materials afterwards by just poking at the entities.

**Reply by prideout** [reply]
Romain is correct but this is an interesting limitation of MaterialProvider that we should think about -- while it lets you provide your own materials, the loader still assumes that you'll provide parameters for the standard glTF parameters.

@jnmaloney If we were to add a new virtual method to MaterialProvider that tells the loader to skip all calls to `setParameter()`, would that solve your problem?  e.g. `bool hasStandardParameters() const = 0`.

Or, do you need a method that lets you opt-out of a subset of standard parameters?

**Reply by jnmaloney** [reply]
@prideout Yes having a flag like `bool hasStandardParameters()` would solve the problem, or alternatively a pair of functions such as `getMaterialInstance` / `createMaterialInstance`

I'm not sure if this will comply with the standard, but is it possible to ignore a parameter if it is not included in the gltf file?
